app/controllers/posts_controller.rb

```diff
@@ -1,14 +1,15 @@
 class PostsController < ApplicationController
-  before_action :set_post, only: [:show, :edit, :update, :destroy]
+  before_action :set_post, only: [:show, :edit, :update, :destroy, :favorite, :unfavorite]
+  before_action :block_unauthorized_changes, only: [:edit, :update, :destroy]
   before_filter :authenticate_user!
   # GET /posts
-  # GET /posts.json
   def index
-    @posts = Post.all.includes(:user)
+    # @posts = Post.all.includes(:user)
+    @posts = Post.where(user_id: current_user.following.map(&:id)).includes(:user)
+    @user_favorites = current_user.favorites.map
```

Comment:
It's bad practice to set more than one instance variable. [source](https://robots.thoughtbot.com/sandi-metz-rules-for-developers)